doc/design/dist_refactor/remote_executor.md

```diff
@@ -0,0 +1,121 @@
+# Design Doc: RemoteExecutor
+## Abstract
+We propose some details to implement `RemoteExecutor`:
+
+- Each `Partitioned IR(intermediate representation)` has unique `IRID`.
+- Each `Partitioned IR` with its relationship with others and its resource need are stored into etcd.
+- Each `PaddlePaddle Runtime` runs `Partitioned IR` got from etcd by `IRID`. 
+  - So the executed `Partitioned IR` can communicate with each other by `IRID` even if some of them restart.
+
+## Architect graph
+<div style="align: center">
+<img src="src/remote_executor2.png" width="700" align=center/>
```

Comment:
I really think that the controller should be controlled by another Python API:

<img width="899" alt="screen shot 2018-01-22 at 11 50 02 am" src="https://user-images.githubusercontent.com/1724178/35241150-69d7912a-ff6a-11e7-9dd8-3dc869498ee9.png">

Reply:
I have one question:
If `RemoteExecutor` restarts, where can `Paddle Runtime` get `Partitioned IR`?From a distribution filesystem?

Reply:
This looks to me that there is one global gatekeeper (RemoteExecutor) to orchestrate all the roles and traffic in and out of the cluster? I think this kind of remote executor should be one per job?